x/ibc/03-connection/keeper/handshake.go

```diff
@@ -0,0 +1,242 @@
+package keeper
```

Comment:
Looks like there are a number of error types to be created here. They should be similar to ICS-02's handshake errors and could likely be ported.